src/ToolInfo.php

```diff
@@ -23,83 +23,114 @@
                     break;
                 }
             }
         }
 
-        return $result;
+        return $this->composerVersion;
     }
 
-    public static function getVersion()
+    public function getVersion()
     {
-        if (self::isInstalledByComposer()) {
-            return Application::VERSION.':'.self::getComposerVersion();
+        if ($this->isInstalledByComposer()) {
+            return Application::VERSION.':'.$this->getComposerVersion();
         }
 
         return Application::VERSION;
     }
 
-    public static function isInstalledAsPhar()
+    public function isInstalledAsPhar()
     {
-        static $result;
-
-        if (null === $result) {
-            $result = 'phar://' === substr(__DIR__, 0, 7);
+        if (null === $this->isInstalledAsPhar) {
+            $this->isInstalledAsPhar = 'phar://' === substr(__DIR__, 0, 7);
         }
 
-        return $result;
+        return $this->isInstalledAsPhar;
     }
 
-    public static function isInstalledByComposer()
+    public function isInstalledByComposer()
     {
-        static $result;
-
-        if (null === $result) {
-            $result = !self::isInstalledAsPhar() && file_exists(self::getScriptDir().self::COMPOSER_INSTALLED_FILE);
+        if (null === $this->isInstalledByComposer) {
+            $this->isInstalledByComposer = !$this->isInstalledAsPhar() && file_exists($this->getComposerInstalledFile());
         }
 
-        return $result;
+        return $this->isInstalledByComposer;
     }
 
-    private static function getScriptDir()
+    private function getComposerInstalledFile()
     {
-        static $result;
-
-        if (null === $result) {
-            $script = $_SERVER['SCRIPT_NAME'];
+        if (null === $this->composerInstalledFile) {
+            $script = $this->script;
 
-            if (is_link($script)) {
+            while (is_link($script)) {
                 $linkTarget = readlink($script);
 
                 // If the link target is relative to the link
                 if (false === realpath($linkTarget)) {
-                    $linkTarget = dirname($script).'/'.$linkTarget;
+                    $linkTarget = dirname($script).DIRECTORY_SEPARATOR.$linkTarget;
                 }
 
                 $script = $linkTarget;
             }
 
-            $result = dirname($script);
+            $composerInstalledJson = DIRECTORY_SEPARATOR.'composer'.DIRECTORY_SEPARATOR.'installed.json';
+            // Windows real file is under `vendor/bin` because composer
+            // doesn't use symlinks on that OS
+            // On Linux instead the executabile is under `vendor/maintainer/package`
+            $this->composerInstalledFile = dirname(dirname($script)).$composerInstalledJson;
+            if (!file_exists($this->composerInstalledFile)) {
+                $this->composerInstalledFile = dirname(dirname(dirname($script))).$composerInstalledJson;
+            }
```

Comment:
Ok, like Ocramius/PackageVersions I would to for `__DIR__` in replacement of `$_SERVER['SCRIPT_NAME']`, commits soon